### PR TITLE
kbailey: potentially malicious SSO DNS lookup

### PR DESCRIPTION
**Pull request by k-bailey**
### Background

Recently there was an active and successful phishing campaign targeting major tech companies using fake SSO/VPN login pages for credential harvesting. This detection attempts to look for the domain name patterns these attackers used in DNS requests.

ref: https://blog.group-ib.com/0ktapus

### Changes

* New rule & config

### Testing

* Make lint/test/fmt
* Data replay using "Panther" as the configured company

### CODE REVIEW

**Review by edyesed — CHANGES_REQUESTED** [review]
I'm interpreting this detection as behaving in a way to fast-exit if p_any_domain_names has anything from the allow list in it. I  suspect this needs to be restructured to return True if  any malicious domain is found vs returning False if any good domain is found.

File: rules/standard_rules/malicious_sso_dns_lookup.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ALLOWED_DOMAINS = [
+    ".amazonaws.com",
+    ".okta.com",
+    ".oktapreview.com",
+    #   "COMPANY.com",
+]
+
+
+def rule(event):
+    # filter out known good domains
```
Comment (edyesed) [inline rules/standard_rules/malicious_sso_dns_lookup.py:39]: something like this, I think 

```python
for domain in event.get("p.any_domain_names", []):
   domain_was_allowed = [x for x in ALLOWED_DOMAINS if domain.lower().endswith(x)]
   if domain_was_allowed:
      continue # loop to next domain in p_any_domain_names
   if COMPANY_NAME in domain.lower():
      fake_matches = [ x for x in FAKE_KEYWORDS if x in domain.lower() ]
      if fake_matches:
          return True
     return False

```

File: rules/standard_rules/malicious_sso_dns_lookup.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        "Entitlements": "15",
+        "RequestType": "1",
+        "event_platform": "Mac",
+        "event_simpleName": "DnsRequest",
+        "name": "DnsRequestMacV2",
+        "p_any_domain_names": [
+        ],
+        "timestamp": "2022-08-31 07:03:49.195"
+      }
+
```
Comment (edyesed) [inline rules/standard_rules/malicious_sso_dns_lookup.yml:80]: We should have a test case where p_any_domain_names has both good and bad domains in it, with the good domain first in the list.

File: rules/standard_rules/malicious_sso_dns_lookup.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+"""
+We HIGHLY recommend running this logic over 30 days of historical data using data replay
+before enabling this in your instance. If ALLOWED_DOMAINS is not fully populated with domains
+you own, that conain your company name, false positive alerts will be generated.
```
Comment (jpl5280) [inline rules/standard_rules/malicious_sso_dns_lookup.py:4]: `conain` -> `contain`

File: rules/standard_rules/malicious_sso_dns_lookup.py (hunk 1 above)
Comment (k-bailey) [reply rules/standard_rules/malicious_sso_dns_lookup.py]: fixed, discussed in person

**Review by edyesed — APPROVED** [review]
this new detection will look for potentially malicious SSO DNS lookups as in the recent oktapus campaign

**Pull request merged**